teku/src/main/java/tech/pegasys/teku/cli/subcommand/RemoteSpecLoader.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.cli.subcommand;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException;
+import tech.pegasys.teku.infrastructure.logging.SubCommandLogger;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.spec.SpecFactory;
+import tech.pegasys.teku.spec.config.SpecConfigLoader;
+import tech.pegasys.teku.sync.forward.singlepeer.RetryDelayFunction;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpClientAuthLoggingIntercepter;
+import tech.pegasys.teku.validator.remote.apiclient.OkHttpValidatorRestApiClient;
+
+class RemoteSpecLoader {
+  private static BiConsumer<Throwable, Integer> retryErrObserver =
+      (Throwable e, Integer attempt) -> {
+        var errMsg = String.format("Attempt %d failed: \n%s\n", attempt, e.getMessage());
+        SubCommandLogger.SUB_COMMAND_LOG.error(errMsg);
+      };
+
+  private static RetryDelayFunction retryDelayFunction =
+      RetryDelayFunction.createExponentialRetry(2f, Duration.ofSeconds(2), Duration.ofSeconds(20));
```

Comment:
We probably don't want to create a dependency on the `sync` packages here so if we really need exponential backoff we should move `RetryDelayFunction` to somewhere like the `infrastructure/async` module.  But I think in this case a constant retry time of something like 5 seconds would make more sense.  It's not an expensive method to serve so there's no real risk of these requests causing the beacon node to be overloaded.

Reply:
Ok, I'll change it to a constant delay instead